src/emailagent.cpp

```diff
@@ -1253,11 +1253,14 @@ void EmailAgent::removeAction(quint64 actionId)
 
 void EmailAgent::saveAttachmentToDownloads(const QMailMessageId messageId, const QString &attachmentLocation)
 {
-    QString attachmentDownloadFolder = QStandardPaths::writableLocation(QStandardPaths::DownloadLocation) + "/mail_attachments/" + attachmentLocation;
     // Message and part structure can be updated during attachment download
     // is safer to reload everything
     const QMailMessage message (messageId);
     const QMailMessagePart::Location location(attachmentLocation);
+    QMailAccountId accountId = message.parentAccountId();
+    QString attachmentDownloadFolder = QStandardPaths::writableLocation(QStandardPaths::DownloadLocation) + "/mail_attachments/"
+            + QString("%1").arg(accountId.toULongLong()) + attachmentLocation;
```

Comment:
Same as above - maybe add a static function to AttachmentListModel to return this path?